backend/services/project_search_service.py

```diff
@@ -83,10 +84,14 @@ def create_search_query():
                 Organisation.name.label("organisation_name"),
                 Organisation.logo.label("organisation_logo"),
             )
-            .filter(Project.private.is_(False))
             .outerjoin(Organisation, Organisation.id == Project.organisation_id)
             .group_by(Organisation.id, Project.id)
         )
+
+        # Remove private projects if user role is not admin.
+        if user is None or user.role != UserRole.ADMIN.value:
```

Comment:
We need to consider here also org manager, project team managers and project authors. Probably using the `is_user_action_permitted_on_project` method of `ProjectAdminService`

Reply:
visibility at that level are planned to be handled in this issue: https://github.com/hotosm/tasking-manager/issues/2817. BTW, AFAIK we agreed to set visibility and permissions to teams level only